src/radiation_system.hpp

```diff
@@ -133,18 +160,68 @@ template <typename problem_t> class RadSystem : public HyperbolicSystem<problem_
 
 	AMREX_GPU_HOST_DEVICE static auto ComputeEddingtonFactor(double f) -> double;
 	AMREX_GPU_HOST_DEVICE static auto ComputePlanckOpacity(double rho, double Tgas) -> double;
+	AMREX_GPU_HOST_DEVICE static auto ComputePlanckOpacityForGroup(double rho, double Tgas) -> quokka::valarray<double, nGroups_>;
```

Comment:
I don't think it makes sense to have both `ComputePlanckOpacity` and `ComputePlanckOpacityForGroup`. I think it should be named `ComputePlanckOpacity` and it should just return an array of the opacities for all of the groups.